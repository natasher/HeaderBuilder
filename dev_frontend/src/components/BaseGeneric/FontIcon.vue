<template>
  <i
    :class="classObject"
    aria-hidden="true">
  </i>
</template>

<script>
export default {

  name: 'FontIcon',

  props: {
    /**
     * Icon name from Font Awesome cataloge.
     */
    icon: {
      type    : String,
      required: true,
    },
  },

  computed: {
    classObject: function () {
      return `icon-${ this.icon }`
    },
  },
}
</script>
