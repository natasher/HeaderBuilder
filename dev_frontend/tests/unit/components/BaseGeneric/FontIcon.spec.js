import { shallowMount } from '@vue/test-utils'
import FontIcon from '../../../../src/components/BaseGeneric/FontIcon.vue'

describe('FontIcon.vue', () => {

  test('should match snapshot', () => {
    const wrapper = shallowMount(FontIcon, {
      propsData: {
        icon: 'poop'
      }
    })

    expect( wrapper.element ).toMatchSnapshot()
  })

  test('should generate class by joining `icon-` with prop with name', () => {
    const icon = 'poop'
    const wrapper = shallowMount(FontIcon, {
      propsData: {
        icon,
      }
    })

    expect( wrapper.classes() ).toContain( `icon-${ icon }` )
  })
})