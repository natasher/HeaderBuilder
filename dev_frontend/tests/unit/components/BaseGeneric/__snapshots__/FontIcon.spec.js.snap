// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontIcon.vue should match snapshot 1`] = `
<i
  aria-hidden="true"
  class="icon-poop"
/>
`;
